### ISSUE TITLE
Polish & release 3.0

### ISSUE THREAD

**Issue by Pathoschild** [Pathoschild/FluentHttpClient#44]
This is just a tracking ticket to finalise the 3.0 release:

* [x] Ensure code conventions are applied, all code is documented, etc.
* [x] Ensure all unit tests pass.
* [x] Polish documentation, release notes, etc.
* [x] Test the client in a few projects.
* [ ] Deploy to NuGet.

**Comment by Jericho** [comment]
Anything I can do to help you go through this checklist?

**Comment by Pathoschild** [comment]
Testing the latest `develop` version in real-world projects would be helpful. Otherwise I just need to sit down and go through it; I'll do that later today if nothing comes up.

**Comment by Jericho** [comment]
I ran all the unit tests and integration tests in my project and everything works fine.

**Comment by Pathoschild** [comment]
I polished the [readme](https://github.com/Pathoschild/FluentHttpClient/tree/develop#readme) and [release notes](https://github.com/Pathoschild/FluentHttpClient/blob/develop/RELEASE-NOTES.md#30); how does that look?

**Comment by Pathoschild** [comment]
Works fine in my projects, so I think we're ready to release.

**Comment by Pathoschild** [comment]
Here's the NuGet package if you want to test before we release:
* [Pathoschild.Http.FluentClient.3.0.0.nupkg.zip](https://github.com/Pathoschild/FluentHttpClient/files/756493/Pathoschild.Http.FluentClient.3.0.0.nupkg.zip)

**Comment by Jericho** [comment]
Reviewing the readme and notes, here are my comments:

**Readme**
1. The 'Custom Retry' section an example that demonstrate how to set the request coordinator. However, the `getDelay` in the sample returns an `int` instead of the expected `TimeSpan`. A better example should be something like:
```CSharp
getDelay: (attempt, response) => return TimeSpan.FromSeconds(attempt) // 1, 2, and 3 seconds
```

2. In order to make it even more obvious that developers can replace the error handler, I suggest adding a sentence similar to:
 _If you have more complex needs regarding error handling (e.g.: you need to parse the body of the response to get a description of errors returned by APIs), please see the 'Custom Filters' section below where we show you how to remove the built-in error handler and how to replace it with your own._ 

3. The last code sample in the 'Custom Filters' section shows a simple custom that sets the `Authentication` header before dispatching every request. However, we recently added convenient extension methods to the client and to the request specifically for this purpose therefore we should show a different sample. Unfortunately, I can't think of another example off the top of my head right now.

4. Similarly to my bullet point 3 above, the code sample in the 'Custom HTTP' section should show something other than setting `Authorization` and `Proxy` since we recently added a convenient constructor which accepts a proxy and we added extension methods for authentication. I can't think of another example at the moment though.

**Release notes**
Looks very good to me!

**Comment by Jericho** [comment]
Downloaded the nuget package: all my unit tests passed and my integration tests succeeded as well

**Comment by Pathoschild** [comment]
Readme changes:

1. Done.
2. I tweaked the section to mention custom filters as an option.
3. I can't think of a better example off-hand. By design all the common scenarios have fluent methods, so this is intended for edge cases that would be too obtuse for a quick example. I added an inline comment to clarify that it's just an example and call out `client.SetAuthentication`. We can change it later if we think of a better example.
4. I changed the example to show how to mock HTTP requests.

How's that?
